### PR TITLE
e2e: promote CRD defaulting test to conformance test

### PR DESCRIPTION
**Pull request by sttts**
/kind feature

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/apimachinery/custom_resource_definition.go [hunk 1]
```diff
@@ -251,6 +257,128 @@ var _ = SIGDescribe("CustomResourceDefinition resources [Privileged:ClusterAdmin
...
 	})
 
+	/*
+		Release : v1.17
+		Testname: Custom Resource Definition, defaulting
+		Description: Create a custom resource definition without default. Create CR. Add default and read CR until
+		the default is applied. Create another CR. Remove default, add default for another field and read CR until
+		new field is defaulted, but old default stays.
+	*/
+	framework.ConformanceIt("custom resource defaulting for requests and from storage works ", func() {
```
Comment (johnbelamaric) [inline test/e2e/apimachinery/custom_resource_definition.go]: If this is a new test it needs to be added as an e2e and soak for a bit before we promote it (can't be flaky). are there any existing e2e tests for this?

Comment (johnbelamaric) [reply test/e2e/apimachinery/custom_resource_definition.go]: I would suggest a separate PR that adds this as an e2e test that uses the beta API. That way we get a signal on flakiness. then the PR to promote to GA would only promote, not implement that test

**Comment by sttts** [comment]
We had soak test time. Should we promote now? @johnbelamaric

**Comment by sttts** [comment]
Here is the test grid history: https://testgrid.k8s.io/sig-release-master-informing#gce-cos-master-default&include-filter-by-regex=custom%20resource%20defaulting&width=5. Pretty green.

**Comment by liggitt** [comment]
promotion lgtm

**Pull request merged**